crates/hir_def/src/body/lower.rs

```diff
@@ -848,8 +853,23 @@ impl ExprCollector<'_> {
                     Pat::Missing
                 }
             }
+            ast::Pat::MacroPat(mac) => match mac.macro_call() {
+                Some(call) => {
```

Comment:
I wonder why is this an option? Feels like it can never be none?

Reply:
I guess the ungrammar codegen makes every sub-node an `Option`